src/storage/gcloud-storage/storage.ts

```diff
@@ -0,0 +1,111 @@
+import type { Storage } from '@google-cloud/storage';
+import type { Account } from 'account/account.entity';
+import type { AccountService } from 'account/account.service';
+import { type Result, error, getError, isError, ok } from 'core/result';
+import type { Context } from 'hono';
+import { v4 as uuidv4 } from 'uuid';
+
+const ALLOWED_IMAGE_TYPES = [
+    'image/jpg',
+    'image/jpeg',
+    'image/png',
+    'image/webp',
+];
+
+export type FileValidationError = 'file-too-large' | 'file-type-not-supported';
+
+export function createStorageHandler(
+    accountService: AccountService,
+    storage: Storage,
+) {
+    /**
+     * Handle an upload to GCloud Storage bucket
+     */
+    return async function handleUpload(ctx: Context) {
+        const logger = ctx.get('logger');
+
+        const formData = await ctx.req.formData();
+        const file = formData.get('file') as File;
+
+        if (!file) {
+            return new Response('No file provided', { status: 400 });
+        }
+
+        const validationResult = validateFile(file);
+
+        if (isError(validationResult)) {
+            const error = getError(validationResult);
+            switch (error) {
+                case 'file-too-large':
+                    logger.error(`File is too large: ${file.size} bytes`);
+                    return new Response('File is too large', { status: 413 });
+                case 'file-type-not-supported':
+                    logger.error(`File type ${file.type} is not supported`);
+                    return new Response(
+                        `File type ${file.type} is not supported`,
+                        {
+                            status: 415,
+                        },
+                    );
+            }
+        }
+
+        const bucketName = process.env.GCP_BUCKET_NAME;
+        const emulatorHost = process.env.GCP_STORAGE_EMULATOR_HOST; // This is for dev and testing environments
+
+        if (!bucketName) {
```

Comment:
I already have this check on boot, but TypeScript complains if I don’t check it again here.
The other option is to use `const bucketName = process.env.GCP_BUCKET_NAME!`;